app/controllers/species_list_controller.rb

```diff
@@ -390,6 +392,61 @@ def destroy_species_list # :norobots:
     end
   end
 
+  def add_remove_observations # :prefetch: :norobots:
+    pass_query_params
+    @id = params[:species_list].to_s
+    @query = find_query(:Observation)
+    if !@query
+      flash_error(:species_list_add_remove_no_query.t)
+      redirect_to(action: "list_species_lists")
+    end
+  end
+
+  def post_add_remove_observations # :prefetch: :norobots:
+    pass_query_params
+    id = params[:species_list].to_s
+    query = find_query(:Observation)
+    species_list = lookup_species_list(id)
+    flash_error(:species_list_add_remove_no_query.t) if !query
```

Comment:
I'd make a small local function that combines find_query and this flash_error.  This could also be reused in add_remove_observations.